include/CppUTest/MemoryLeakDetector.h

```diff
@@ -216,7 +216,11 @@ class MemoryLeakDetector
     void removeMemoryLeakInformationWithoutCheckingOrDeallocatingTheMemoryButDeallocatingTheAccountInformation(TestMemoryAllocator* allocator, void* memory, bool allocatNodesSeperately);
     enum
     {
+#ifdef CPPUTEST_DISABLE_MEM_CORRUPTION_CHECK
+        memory_corruption_buffer_size = 0
+#else
         memory_corruption_buffer_size = 3
```

Comment:
I guess here you also want to do the sizeof(guardbyte) ?

Reply:
I guess by making GuardBytes empty when disabled, you do not require this variable at all and most of the #ifdef can be removed?

Reply:
Ah, I can't do this as the Microsoft tool-chain doesn't allow zero length arrays.